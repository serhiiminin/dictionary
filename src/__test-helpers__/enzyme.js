import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
