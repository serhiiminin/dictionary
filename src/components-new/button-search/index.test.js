import React from 'react';
import { shallow } from 'enzyme';
import ButtonSearch from '.';

describe('Button control', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ButtonSearch>Text</ButtonSearch>);
  });

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('isActive', () => {
    wrapper.setProps({ isActive: true });
    expect(wrapper).toMatchSnapshot();
  });
});
