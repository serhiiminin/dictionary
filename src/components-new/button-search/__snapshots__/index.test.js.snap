// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button control isActive 1`] = `
<ButtonMenu
  classes={
    Object {
      "buttonMenu": "ButtonMenu-buttonMenu-0-1-1",
    }
  }
  isActive={true}
>
  Text
</ButtonMenu>
`;

exports[`Button control render 1`] = `
<ButtonMenu
  classes={
    Object {
      "buttonMenu": "ButtonMenu-buttonMenu-0-1-1",
    }
  }
>
  Text
</ButtonMenu>
`;
